homeassistant/components/cert_expiry/sensor.py

```diff
@@ -87,15 +82,6 @@ def available(self):
         """Icon to use in the frontend, if any."""
         return self._available
 
-    async def async_added_to_hass(self):
-        """Once the entity is added we should update to get the initial data loaded."""
-
-        def do_update(_):
-            """Run the update method when the start event was fired."""
-            self.update()
-
-        self.hass.bus.async_listen_once(EVENT_HOMEASSISTANT_START, do_update)
```

Comment:
What was the reason for the old behavior before the move to config entries to update only on home assistant start? Was it to be able to check a certificate that is dependent on home assistant finishing setup?

Reply:
https://github.com/jjlawren/home-assistant/blob/c2f1c4b981a173b3e8953ddc92b02d28df7c1959/homeassistant/components/cert_expiry/sensor.py#L33-L46

Reply:
I'm thinking the reason why some users can't import the config is that we check the certificate during import before home assistant has finished setting up. That won't work if home assistant is serving as host.

The other problem is the two close updates that we're trying to fix here. Not sure what's the best way. Either it will work after a restart but not directly after creating the integration. Or it will work directly after creating the integration but not after a restart.

Maybe we should create the entity as available and just update on start?

Reply:
I couldn't say. I tested with this new change quite a few times and it didn't introduce any new issues. The entities were added reliably each time.